main/mainworker.cpp

```diff
@@ -12885,59 +12885,57 @@ void MainWorker::HeartbeatCheck()
 					}
 				}
 			}
+		}
 
-			if (pHardware->m_DataTimeout > 0)
+		if (pHardware->m_DataTimeout > 0)
+		{
+			//Check Receive Timeout
+			double diff = difftime(now, pHardware->m_LastHeartbeatReceive);
+			if (diff > pHardware->m_DataTimeout)
 			{
-				//Check Receive Timeout
-				double diff = difftime(now, pHardware->m_LastHeartbeatReceive);
-				if (diff > pHardware->m_DataTimeout)
+				std::vector<std::vector<std::string> > result;
+				result = m_sql.safe_query("SELECT Name FROM Hardware WHERE (ID='%d')", pHardware->m_HwdID);
+				if (result.size() == 1)
```

Comment:
This should always be 1... can't be different ;) for this reason, please change this to if (!result.empty())

Reply:
I didn't change anything in this part of the code ;) Just moved that block out of the if !m_SkipReceiveCheck scope. But will change, no problem ;)

Reply:
Maybe you can change it? Screwing things up here with staging and stuff and can't get a good new commit for this branch somehow without committing all changes I made (which is more than this ;)) Haven't use this system for a while..